### DISCUSSION TITLE
Consider using microsoft/STL

### DISCUSSION THREAD

**Discussion by adamyg** [open-watcom/open-watcom-v2#598] [Ideas]
MSVC 2019+ STL is now open-source, consider migrating.
https://github.com/microsoft/STL

**Comment by pchapin** [comment]
The Open Watcom C++ compiler isn't able to compile a full featured STL, so any modern STL implementation would require a considerable amount of conditional compilation to bypass the parts using unsupported C++ features. This isn't necessarily a reason to not port a third part STL to Open Watcom, but people should be aware that it wouldn't increase the amount of supported STL components over OWSTL as much as one might suppose.

That said, there are certainly STL components that Open Watcom could support, but that are not currently in OWSTL. But this raises the question: would it be better to extend OWSTL or execute a non-trivial port of a third party STL?

I should say that I have not actually looked at the MSVC 2019 STL mentioned above, so it is possible that it already has controls for bypassing components based on C++ features required, and so a port to Open Watcom might be easier than I'm imagining. My comments above are informed by my experience in the past with other third party STLs.

**Comment by revelator** [comment]
Hmm at one point it was able to compile and use stlport though with the watcom stl this is no longer possible atleast not without a lot of hacking about to replace some functionality the watcom stl now provides.

stlport itself is also a bit of a hack used to add stl support to compilers which originally lacked that functionality and had much the same problem as watcom. Digital Mars was one such compiler and now uses stlport for most and adds in some parts of apaches libc++ as well as stlsoft. Digital mars works very well with stlport though due to the age it has restrictions on what is possible.
